crates/ghost_actor/src/ghost_channel.rs

```diff
@@ -0,0 +1,271 @@
+use crate::{GhostCallback, GhostResult, GhostTracker, RequestId};
+use std::any::Any;
+
+enum GhostChannelMessage<Request, Response, Error> {
+    Request {
+        request_id: Option<RequestId>,
+        payload: Request,
+    },
+    Response {
+        request_id: RequestId,
+        payload: Result<Response, Error>,
+    },
+}
+
+pub struct GhostMessage<RequestToSelf, RequestToOther, RequestToSelfResponse, Error> {
+    request_id: Option<RequestId>,
+    payload: Option<RequestToSelf>,
```

Comment:
I'm wondering if we shouldn't call this something other than `payload` to distinguish it from the payloads of the GhostActor Request/Response quads.  I was finding that confusing.  It's almost like it should be called `message`, and the generics should be `MsgToSelf`, `MsgToOther`, `MsgToSelfResponse`  which in a way unites in the channel the  Request/Response pairs into a conceptual bundle.